### PR TITLE
Configure docs.bb.net

### PR DESCRIPTION
**Pull request by sa2ajj**
[TRAC-2857](http://trac.buildbot.net/ticket/2857)

### CODE REVIEW

File: jail-docs.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  sudo: yes
+  vars:
+    env_name: doctools
+    web_host_name: "docs.buildbot.net"
+    bbdocs_dir: "bbdocs"
+
+  roles:
+  - base
+  - role: user
+    user_id: "{{ worker_account }}"
```
Comment (djmitche) [inline jail-docs.yml]: I don't think this variable is ever defined?

File: jail-docs.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    repo_url: git://github.com/buildbot/buildbot.git
+    repo_branch: master
+    target_user: "{{ worker_account }}"
+    target_dir: "{{ getent_passwd[worker_account].4 }}/buildbot"
+    target_commands:
+    # `command` module does not allow to set env variables, nor expands them,
+    # so let's use make's variables.
+    - "make -C master/docs PATH:={{ getent_passwd[worker_account].4 }}/{{ env_name }}/bin:$(PATH) VERSION=latest"
+    - "find master/docs/_build/html -name '*.html' -exec python2 {{ getent_passwd[worker_account].4 }}/{{ bbdocs_dir }}/add-tracking.py {} ;"
+    - "ln -sf {{ getent_passwd[worker_account].4 }}/buildbot/master/docs/_build/html {{ getent_passwd[worker_account].4 }}/{{ bbdocs_dir }}/docs/latest"
```
Comment (djmitche) [inline jail-docs.yml]: This is nice - we can eliminate the uploads from the 'docs' builder in metabuildbot.

File: roles/dns/templates/buildbot.net [hunk 3]
```diff
@@ -8,11 +8,11 @@
 @               7200    IN      NS      {{ secondary }}
 {% endfor %}
 buildbot        7200    IN      CNAME   ds0210.flosoft-servers.net.
-docs            7200    IN      CNAME   ds0210.flosoft-servers.net.
 meetings        43200   IN      CNAME   ds0210.flosoft-servers.net.
 xy2dwhhhx5d4    7200    IN      CNAME   gv-d47fzorbndt4i4.dv.googlehosted.com.
 mx              86400   IN      A       140.211.10.235
 @               86400   IN      MX      10 mx.buildbot.net.
+docs            86400   IN      A       140.211.10.237
```
Comment (djmitche) [inline roles/dns/templates/buildbot.net]: I don't think you wanted to stick this here, as it changes the meaning of the `@` below

**Comment by djmitche** [comment]
Let's get this merged without the DNS change, and verify with local /etc/hosts modifications -- but I think it looks fine aside from the missing variable.

File: jail-docs.yml (hunk 1 above)
Comment (djmitche) [inline jail-docs.yml]: oh, I see that it's global, sorry..

**Comment by djmitche** [comment]
So, yeah, good to merge without the DNS change.

File: roles/dns/templates/buildbot.net (hunk 3 above)
Comment (sa2ajj) [inline roles/dns/templates/buildbot.net]: The meaning of `@` is the same throughout the zone.

Comment (djmitche) [inline roles/dns/templates/buildbot.net]: Of course - I'm thinking of a blank.  Still, it's odd to have @ interspersed.

Comment (sa2ajj) [inline roles/dns/templates/buildbot.net]: "Odd" might be a bit too strong word, but as we agreed, this PR will not change the zone.

**Comment by sa2ajj** [comment]
This looks "final" unless there're some other concerns.

**Pull request merged**